adam/learner/objects.py

```diff
@@ -705,7 +697,13 @@ def templates_for_concept(self, concept: Concept) -> ImmutableSet[SurfaceTemplat
         raise RuntimeError(f"Invalid concept {concept}")
 
     def log_hypotheses(self, log_output_path: Path) -> None:
-        pass
+        for concept, hypothesis in self.concepts_to_patterns().items():
+            hypothesis.render_to_file(
+                graph_name="perception",
+                output_file=Path(
+                    log_output_path / f"{str(type(self))}-{concept.debug_string}"
+                ),
+            )
```

Comment:
I think this function is designated for abstract implementation given learners could theoretically store internal concepts to patterns which we could also need to display?

Reply:
We need this to see what objects look like when we use object recognizer